subsys/bluetooth/host/gatt.c

```diff
@@ -3523,16 +3522,23 @@ bool bt_gatt_is_subscribed(struct bt_conn *conn,
 		return false;
 	}
 
-	ccc = attr->user_data;
+	if (!attr->read) {
+		LOG_ERR("Read method not set");
+		return false;
+	}
 
-	/* Check if the connection is subscribed */
-	for (size_t i = 0; i < BT_GATT_CCC_MAX; i++) {
-		const struct bt_gatt_ccc_cfg *cfg = &ccc->cfg[i];
+	len = attr->read(conn, attr, &ccc_bits, sizeof(ccc_bits), 0);
+	if (len < 0) {
+		LOG_ERR("Failed to read attribute (err %zd)", len);
```

Comment:
I think we should print the pointer to the attribute. It could be useful for debugging.

```suggestion
		LOG_ERR("Failed to read attribute %p (err %zd)", attr, len);
```